File: Visuals/Lua/Closest pair of points.lua
-- Code copied from https://commons.wikimedia.org/wiki/File:Closest_pair_of_points.svg

local WD, HT, NUM_POINTS = 256, 256, 16
local COLOR = { normal = '#000', closest = '#f00' }
local RADIUS = 5

math.randomseed(os.time())
local P = {}
for _ = 1, NUM_POINTS do
    P[#P+1] = { x = RADIUS + (WD - 2*RADIUS) * math.random(),
                y = RADIUS + (HT - 2*RADIUS) * math.random() }
end

-- Find closest pair, using naive algorithm.
local closest_a, closest_b
local min_dist
for i, p in ipairs(P) do
    for j, q in ipairs(P) do
        if i ~= j then
            local dist = math.sqrt((p.x - q.x)^2 + (p.y - q.y)^2)
            if not min_dist or dist < min_dist then
                min_dist = dist
                closest_a, closest_b = i, j
            end
        end
    end
end

if min_dist < RADIUS then
    error("points on top of each other, run me again")
end

io.write('<?xml version="1.0" encoding="UTF-8" standalone="no"?>\n',
         '<svg viewBox="0.00 0.00 ', WD, '.00 ', HT, '.00" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink"',
         ' width="', WD, '" height="', HT, '">\n')
for i, p in ipairs(P) do
    local color = COLOR.normal
    if i == closest_a or i == closest_b then color = COLOR.closest end
    io.write(' <circle cx="', RADIUS + p.x, '" cy="', RADIUS + p.y,
             '" r="', RADIUS, '" style="fill:', color, '"/>\n')
end
io.write('</svg>\n')

File: visuals/C++/Closest pair of points.cpp
#include <bits/stdc++.h>
#define point pair<double, double>
using namespace std;
static char *color[] = {"#000", "#f00"};

double randDouble(double low, double high){
	return low + (high - low) * (random() % 100000) / 100000;
}
double euclid(point a, point b){
	a.first -= b.first;
	a.second -= b.second;
	return sqrtl(a.first * a.first + a.second * a.second);
}
int main () {
	ios_base::sync_with_stdio(0);
	cin.tie(0);
	srand(time(NULL));
	int height, width, numPoints;
	tie(height, width, numPoints) = tuple<int, int, int>(256, 256, 16);	
	int radius(5);

	vector<point> p;
	int closest_a = 0, closest_b = 1;
	while(true){
		closest_a = 0, closest_b = 1;
		p.clear();
		for(int i = 0; i < numPoints; ++i){
			p.emplace_back(randDouble(0, height), randDouble(0, width));
		}
		sort(p.begin(), p.end());
		for(int i = 0; i < numPoints; ++i){
			for(int j = i + 1; j < numPoints; ++j){
				if(euclid(p[i], p[j]) < euclid(p[closest_a], p[closest_b])) tie(closest_a, closest_b) = tuple<int, int>(i, j);
			}
		}
		if(euclid(p[closest_a], p[closest_b]) <= radius * 2) {
			printf("Error when building image, try again\n");
		}else break;
	}
	printf("<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"no\"?>\n\
<svg viewBox=\"0.00 0.00 256.00 256.00\" xmlns=\"http://www.w3.org/2000/svg\" xmlns:xlink=\"http://www.w3.org/1999/xlink\" width=\"%d\" height=\"%d\">\"\n", width, height);
	for(int i = 0; i < numPoints; ++i){
		printf("<circle cx=\"%lf\" cy=\"%lf\" r=\"%d\" style=\"fill:%s\"/>\n", p[i].first, p[i].second, radius, color[(i == closest_a || i == closest_b)]);
	}
	printf("</svg>\n");
	return 0;
}